gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2

```diff
@@ -252,6 +259,44 @@ def test__get_universe_domain():
         {{ service.client_name }}._get_universe_domain("", None)
     assert str(excinfo.value) == "Universe Domain cannot be an empty string."
 
+@pytest.mark.parametrize("error_code,cred_info_json,show_cred_info", [
+    (401, CRED_INFO_JSON, True),
+    (403, CRED_INFO_JSON, True),
+    (404, CRED_INFO_JSON, True),
+    (500, CRED_INFO_JSON, False),
+    (401, None, False),
+    (403, None, False),
+    (404, None, False),
+    (500, None, False)
+])
+def test__add_cred_info_for_auth_errors(error_code, cred_info_json, show_cred_info):
+    cred = mock.Mock(["get_cred_info"])
+    cred.get_cred_info = mock.Mock(return_value=cred_info_json)
+    client = {{ service.client_name }}(credentials=cred)
+    client._transport._credentials = cred
+
+    error = core_exceptions.GoogleAPICallError("message", details=[])
+    error.code = error_code
+
+    client._add_cred_info_for_auth_errors(error)
+    if show_cred_info:
+        assert error.details == [CRED_INFO_STRING]
```

Comment:
Can we also add a test to ensure that error details that we receive from the API are not clobbered? 

As an example, we could update the `test_error_details` showcase test which already has tests with error details 
https://github.com/googleapis/gapic-generator-python/blob/main/tests/system/test_error_details.py